transports/webrtc/src/transport.rs

```diff
@@ -0,0 +1,697 @@
+        self.dial(addr)
+    }
+
+    fn address_translation(&self, server: &Multiaddr, observed: &Multiaddr) -> Option<Multiaddr> {
+        libp2p_core::address_translation(server, observed)
+    }
+}
+
+/// A stream of incoming connections on one or more interfaces.
+pub struct WebRTCListenStream {
+    /// The ID of this listener.
+    listener_id: ListenerId,
+
+    /// The socket address that the listening socket is bound to,
+    /// which may be a "wildcard address" like `INADDR_ANY` or `IN6ADDR_ANY`
+    /// when listening on all interfaces for IPv4 respectively IPv6 connections.
+    listen_addr: SocketAddr,
+
+    /// The IP addresses of network interfaces on which the listening socket
+    /// is accepting connections.
+    ///
+    /// If the listen socket listens on all interfaces, these may change over
+    /// time as interfaces become available or unavailable.
+    in_addr: InAddr,
+
+    /// A `RTCConfiguration` which holds this peer's certificate(s).
+    config: RTCConfiguration,
+
+    /// The UDP muxer that manages all ICE connections.
+    udp_mux: Arc<UDPMuxNewAddr>,
+
+    /// Future that drives reading from the UDP socket.
+    udp_mux_read_fut: Option<BoxFuture<'static, UDPMuxEvent>>,
+
+    /// `Keypair` identifying this peer
+    id_keys: identity::Keypair,
+
+    /// Set to `Some` if this [`Listener`] should close.
+    /// Optionally contains a [`TransportEvent::ListenerClosed`] that should be
+    /// reported before the listener's stream is terminated.
+    report_closed: Option<Option<<Self as Stream>::Item>>,
+}
+
+impl WebRTCListenStream {
+    /// Constructs a `WebRTCListenStream` for incoming connections.
+    fn new(
+        listener_id: ListenerId,
+        listen_addr: SocketAddr,
+        config: RTCConfiguration,
+        udp_mux: Arc<UDPMuxNewAddr>,
+        id_keys: identity::Keypair,
+    ) -> Self {
+        let in_addr = InAddr::new(listen_addr.ip());
+
+        WebRTCListenStream {
+            listener_id,
+            listen_addr,
+            in_addr,
+            config,
+            udp_mux,
+            udp_mux_read_fut: None,
+            id_keys,
+            report_closed: None,
+        }
+    }
+
+    /// Report the listener as closed in a [`TransportEvent::ListenerClosed`] and
+    /// terminate the stream.
+    fn close(&mut self, reason: Result<(), Error>) {
+        match self.report_closed {
+            Some(_) => debug!("Listener was already closed."),
+            None => {
+                // Report the listener event as closed.
+                let _ = self
+                    .report_closed
+                    .insert(Some(TransportEvent::ListenerClosed {
+                        listener_id: self.listener_id,
+                        reason,
+                    }));
+            }
+        }
+    }
+
+    /// Poll for a next If Event.
+    fn poll_if_addr(&mut self, cx: &mut Context<'_>) -> Option<<Self as Stream>::Item> {
```

Comment:
```suggestion
    fn poll_if_addr(&mut self, cx: &mut Context<'_>) -> Poll<<Self as Stream>::Item> {
```

Strictly speaking, `Poll` and `Option` are the same, though semantically they transport different meanings. Would it not make more sense to use `Poll` here?

Reply:
Changed it to `Poll`.